src/main/java/net/masterthought/cucumber/json/Embedding.java

```diff
@@ -63,28 +64,35 @@ public String getFileId() {
     }
 
     public String getExtension() {
-        switch (mimeType) {
+        // prepare/ensure switch-case matching
+        String mime = mimeType;
+        // remove subtype's suffix (if existing)
+        if (mime.contains("+")) mime = mime.substring(0, mime.indexOf('+'));
+        // remove subtype's parameter (if existing)
+        if (mime.contains(";")) mime = mime.substring(0, mime.indexOf(';'));
+        // normalise
+        mime = mime.toLowerCase(Locale.ENGLISH).trim();
```

Comment:
why?

Reply:
As specified by RFC-2045 (https://tools.ietf.org/html/rfc2045), "[...] All media type values, subtype values, and parameter names as defined are case-insensitive. [...]".
Thus, there might by upper-case letters and in order to match those cases too, it is converted to lowercase with respect to the english language.

Reply:
In addition, the `mime.substring(0, mime.indexOf(';'))` ensures that the MIME-type's parameter is removed. This ensures that, for example, "text/html; charset=UTF-8" will be simplified as "text/html" in order to match the according case in the switch statement.